src/SharedMemory.php

```diff
@@ -26,7 +26,7 @@ public function __construct()
         # Avoid creating 2 instances using the same memory segment
         usleep(1000);
 
-        $this->key = (int) (microtime(true) * 1000);
+        $this->key = (int) (microtime(true) * 1000) - random_int(1, 999);
```

Comment:
Can you explain more about the problem and why you've approached the fix in this way?

Reply:
i think mayby it creating 2 instances using the same memory segment, and this commit fixed it problem on our production server.

Reply:
Hmm ok, I can see how that could happen, in which case we should remove the `usleep()` above.

But also it would probably be safer to always append the random number to the end, rather than adjusting the timestamp, otherwise the following would still give duplicates:
1593524164.4012 - random 3012 = 15935241641000
1593524164.5024 - random 4024 = 15935241641000
Vs appending:
1593524164.4012 . random 3012 = 159352416440123012
1593524164.5024 . random 4024 = 159352416450244024

What do you think?

Reply:
yes, we can rm usleep. about concat i test and error still exists.

Reply:
echo (int) (microtime(true) * 1000) - random_int(1, 999) . PHP_EOL; echo (int) (microtime(true) * 1000) - random_int(1, 999) . P
HP_EOL; ";
1593525307693
1593525307786